### ISSUE TITLE
Difficult to configure if installed using the msi installer

### ISSUE THREAD

**Issue by buro9** [prometheus-community/windows_exporter#976]
I've installed the windows_exporter using the released msi package (amd64) and it has done everything I've expected... created `C:\Program Files\windows_exporter` and put the files in there and also created a Windows Service called `windows_exporter`.

However if I now wish to edit the default configuration, i.e. to provide command line flags to include `memory`, then this is now difficult without a registry edit or reinstall.

What happened:
* I couldn't configure the Service to have the extra command line args without using Registry Editor to edit `Computer\HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\windows_exporter\Image Path`
* I couldn't utilise the config file option as the command line args default to not expecting it ( https://github.com/prometheus-community/windows_exporter/blob/master/exporter.go#L261-L264 has no default value for the config file )
* I couldn't utilise any environment variables (the flags dependency, kingpin, does not appear to consider environment variables as a source for config)
* I reread the README and see that my only way to reconfigure is to reinstall and provide the args to `msiexec`

I resorted to the registry edit.

What I expected:

To be able to install using a released msi, and then to trivially reconfigure post-install.

In other words, I expected one of the following to be documented and to work:

* To be able to configure a .yml config file that was already being looked at
* Or to be able to configure a default run file that would contain the executable args
* Or to be able to configure the executable using environment variables
* And for the readme to describe which is the preferred (supported) way to do this (even if the supported way turns out to be "edit your Registry")

**Comment by trippinnik** [comment]
I don't bother with the msi. I just grab the exe and configure the service.

**Comment by breed808** [comment]
There's two intended methods of use for the exporter.

First is the MSI, which is intended to be deployed by a tool such as SCCM, deploying a service and firewall rule out-of-the-box. As no configuration management is assumed, exporter arguments are statically defined in the service definition. Although these can be changed, it can be a painful process.

Second is the EXE, which does not configure a service or firewall entry. The EXE is expected to be used with some form of configuration management (Ansible, Salt, Puppet, etc) to manage the firewall and arguments/config/service.

I can't recall if we've previously discussed including a default config, but I don't think it'd be helpful in the first scenario (MSI deployment via SCCM), as a config management tool would still be required to manage the configuration file.

We've not considered environment variables before. It may be worth discussing that as an option if command arguments and config files are inadequate.

**Comment by FLeven** [comment]
I deploy the MSI with chocolatey, pass the path to the config yml with it and reconfigure the exporter with the yml file I necessary and restart the service. I could also deploy the yml file by group policy. I got many problems with the exporter, but since I use the config file, configuration is not one of them anymore.
